### PR TITLE
dev-cmd/contributions: CSV output of totals per maintainer

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

- Part of https://github.com/Homebrew/brew/issues/13642.
- Turns out in my head a few days ago I was overcomplicating this. I had a brainwave while in the shower.
- Some refactoring so that we call `totals` to sum up the hash of hashes less, since the grand total numbers are now used in multiple places.

### CODE REVIEW

File: Library/Homebrew/dev-cmd/contributions.rb [hunk 1]
```diff
@@ -114,7 +129,7 @@ def generate_csv(user, results)
           counts.values.sum,
         ]
       end
-      csv << [user, "*", "*", "*", "*", total(results)]
+      csv << [user, "*", "*", "*", "*", grand_total]
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/contributions.rb:132]: Might be nice to have each of the `count` columns have values here:

```suggestion
      csv << [user, "*", "commits_total", "coauthorships_total", "signoffs_total", grand_total]
```

Doesn't need to block this PR, though!

**Review by MikeMcQuaid — APPROVED** [review]
Makes sense to me!

File: Library/Homebrew/dev-cmd/contributions.rb (hunk 1 above)
Comment (issyl0) [reply Library/Homebrew/dev-cmd/contributions.rb]: Done! Output now (for a single user):

```csv
user,repo,commits,coauthorships,signoffs,total
issyl0,brew,363,13,0,376
issyl0,core,473,24,327,824
issyl0,cask,4,0,0,4
issyl0,aliases,0,0,0,0
issyl0,autoupdate,1,0,0,1
issyl0,bundle,17,2,0,19
issyl0,command-not-found,1,0,0,1
issyl0,test-bot,3,0,0,3
issyl0,services,9,0,0,9
issyl0,cask-drivers,0,0,0,0
issyl0,cask-fonts,0,0,0,0
issyl0,cask-versions,0,0,0,0
issyl0,all,871,39,327,1237
```

Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/contributions.rb]: Yay, thanks @issyl0!

**Pull request merged**